### DISCUSSION TITLE
Test of GOODF

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/GoOdf#32] [General]
Hello all,

I've tried to create an organ with GOODF today .. the fact is I have spent 2-3 hours to understand the program, try to export to GO, close GOODF, reopened it as I saw my organ didn't work and see ... all my 2-3 hours work was lost 😢 
GOODF opened completely empty ..

Maybe I haven't understood the program but it is quite hard to use .. well not hard but we have to enter in the logic of organ building. 
Is it right we have to enter 4 times the composition ???
- in manuals
- in ranks
- in switches
- in main panel

We have some récurent options, as "create new pipes by MIDI" in Ranks tab, in Stops tab ... it is an example but it is not intuitive ...

So, I've created my panel but when I import in GO, I don't have any sound except one time just the continuous sound of the first C note when I select the stop ...

I have the impression to follow each step but without understanding anything.. 😕

**Comment by larspalo** [comment]
Thanks for testing GOODF! Yes, I can understand that it might be a bit overwhelming at first to use GOODF. Since my goal was to make everything in the odf possible to edit in the program it's also pretty complex.

Explaining the process of using the program in words would be extremely time consuming so therefore I've created a screencast video series as a separate playlist on my (GrandOrgue related) YouTube channel. You can find it at https://www.youtube.com/playlist?list=PLNz9AJhjubTdw0pV2nZC_XfphJhJV5ly3 to watch a few tutorials that I've recorded. More will come later.

As I briefly note in the first overview - you'll need to have "some" knowledge about the samples you're using to make the right decisions while creating the organ, but I think this should get you started!

**Reply by Jean-Gabriel1982** [reply]
Dear Lars,
Ohh thanks you for the screencasts, I'm watching the second one and it is so much clear now !!
I'll continue testing GOODF of course 😄

**Comment by Bumblebee001** [comment]
I have been able to use GOODF out of the box so to speak with few stumbling
blocks because I am accustomed to writing odfs from scratch and I have also
used Vincent Foreman’s organ builder and that was a simple but incomplete
software. So yes it does help to have some prior knowledge of odfs. Also it
is important to know how samples need to be processed (with markers and
loops and releases) before you can incorporate them into your sample-set
otherwise it just would not work!

Mark
Maltese Historic Pipe Organs

**Reply by Jean-Gabriel1982** [reply]
@Bumblebee001 thank you for your message, I understand we need to have knowledge on odf files but I hope it is not an obligation to use goodf ..
The main thing to do is to understand the structure of the organ and,

@larspalo about structure, i watched on your screencast 2 that you began by windchest, after go to manuals, then ranks, accessories (trémulant) then manuals ...

- is there a way to implement step by step system, for example you have a warning when you begin by manuals for example, telling you have first to create windchest ... the goal is to "force" users to understand the structure (even if I know I'm alone to misunderstood 😄 )

- for accessories (tremulant, swell box) is it possible in the futur to create within the manuals tab (like stops, divisional ...) as we cas avoid creating those accessories and after designate to a specific keyboard ?

I'll continue testing .... 😀

**Comment by larspalo** [comment]
Eventually I'll add some odf logic validation and warnings if the current organ is incomplete for GrandOrgue usage, but that's not main priority at the moment.

The exact order one creates things is not extremely crucial as long as all the necessary elements exist and proper references between them are made. Because of that, and to keep the flexibility to a maximum, I don't think that I'll add a step by step wizard in the close future, and even if it would be implemented, I'd certainly not make it mandatory to use - more of an optional help if one wants it.

Tremulants and swell boxes are not necessarily tied only to one manual like the stops, couplers and divisionals are. They can affect multiple manuals (or rather divisions). That's why they are organ wide elements.

**Reply by Jean-Gabriel1982** [reply]
Ok but for example if I put stops within the tab "Manual" I don't need to recreate them in "Rank" tab ?

My "problem" is to go to a tab, then go another, then come back to the first tab to confirm the second tab etc ... 
For confirmed users it is automatic but for new ones it is a bit labyrinth 🥺

**Reply by larspalo** [reply]
A "stop" can have an internal rank (a rank not defined separately but in the stop itself) and then you don't need to create any ranks at all separately. Ranks are more or less only needed if you create composites of multiple ranks within a stop (like a manually created mixture or other compound stop), you want to borrow or re-use ranks or parts of them among multiple stops or any similar situation. Otherwise, they are totally optional and not needed as separate elements at all. For many (most) organs you can safely ignore ranks completely if you want to as there are other methods to achieve the same results.

That some create a rank separately and then reference that single rank in a single stop in only a single instance could perhaps at best be considered a "correct" logical approach, but it's totally unnecessary if no special reason (like I mentioned above) exist.

**Reply by Jean-Gabriel1982** [reply]
Thank you for all those explications !

Other question, is it possible to add custom switches in place the 13 possibilities we have actually ?

Possible to add foot pistons too ?

**Reply by larspalo** [reply]
> is it possible to add custom switches in place the 13 possibilities we have actually ?

I'm not sure I understand what you actually mean, but it's certainly possible to create switches if you have any use for them. I think that switches will be one of my next topics for the screencast series. The bottom line will be to have a clear use case for needing them, otherwise they are not necessary.

> Possible to add foot pistons too ?

Well, certainly! The method to do it depends on exactly what you mean by foot pistons (what you want to do with them), but everything's possible. How to use the divisionals and generals (and when to perhaps use the built in versions rather than the older odf style) is definitely another topic for a screencast.

**Reply by Jean-Gabriel1982** [reply]
Sorry I mistaken .. for stops in Panel tab > GUI elements > Stop 

We can chose in a list of DispImageNum 7 (I called them switches but I'm probably wrong) .. I just want to know if it is possible to add our own button ..

Regarding foot pistons, it is for couplers I/P, II/P etc ...

**Reply by larspalo** [reply]
Yes, you can add your own custom graphics that will be used instead of the built in versions. Screencast number 4 is about that. To tell the truth, anything (more or less) could be displayed as anything you can imagine and create!

**Reply by Jean-Gabriel1982** [reply]
Ah i have watched screencast 4 but not completely I think .. I saw how you customise manuals and pedal and put the Stops buttons on the background image but I haven't seen how to change de stops buttons ..

**Reply by larspalo** [reply]
> but I haven't seen how to change de stops buttons

All type of buttons, switches and drawstops are treated the same. You just add your custom on and off image for them and now you have changed from the built in graphics to your own custom.

**Reply by Jean-Gabriel1982** [reply]
Ok this evening I tried to make again a little organ .. I think I arrive in a good result, but I have a problem with sound ..

I've just put a stop Montre 8' on one manual, but when I play it on GO when I stop playing, stay one note during 2-3 seconds even if I played 8 notes at a time ..

**Reply by larspalo** [reply]
> but when I play it on GO when I stop playing, stay one note during 2-3 seconds even if I played 8 notes at a time

I'd suspect that there is some problem with how you've setup the releases for the samples you're using. Many (attack/sustain) samples doesn't contain any release in them (most certainly the case if you're using samples from SP or similar vendor), and if so you should not try to load any release from the attack samples either. You do this by un-checking the box for "Load release in attack".

![LoadReleaseOption](https://github.com/larspalo/GOODF/assets/6173941/8fa558e1-4f5a-4f3c-a1de-ab182156d0d3)

You can also fix this without re-loading all the pipes by selecting first attack in the first troublesome pipe and edit its attack properties to not load release, then copy that to all other attacks coming from the same directory.

![CopyAttackProperties](https://github.com/larspalo/GOODF/assets/6173941/e2c75fb9-d367-4270-b416-8a41b1102a22)

All this boils down to that one needs to know how the samples are intended to be used to load them properly... LoopAuditioneer might be a nice friend to get to know so that the samples could be examined.

**Reply by Jean-Gabriel1982** [reply]
Ok I'll try to do that I think tomorrow... hard day at work today 🥲 

Other thing, I have, for swell enclosure, 2 folders for each stop, right and left. 

To import, I have to select "create new pipes by midi..." and for the second folder "add more samples from" ?

**Reply by larspalo** [reply]
> Other thing, I have, for swell enclosure, 2 folders for each stop, right and left.

It all depends on exactly what you want to achieve. I've no idea of what samples you're using, how they were intended to be used, or how you actually want to use them.

If you have surround samples, it might pay off to create separate windchests for the front and rear ranks separately so that they easily can be mixed at will by the user. But I don't know the details of what your're using, please understand that there is no universal standard of sample sets or how individual samples are done or best utilized.

One of the real beauties with GO is that you are free to do what you want with it, but it also requires you to have the knowledge to do it, once you step outside the basic pre-defined user role.

**Reply by Jean-Gabriel1982** [reply]
> > but when I play it on GO when I stop playing, stay one note during 2-3 seconds even if I played 8 notes at a time
> 
> 
> 
> I'd suspect that there is some problem with how you've setup the releases for the samples you're using. Many (attack/sustain) samples doesn't contain any release in them (most certainly the case if you're using samples from SP or similar vendor), and if so you should not try to load any release from the attack samples either. You do this by un-checking the box for "Load release in attack".
> 
> 
> 
> ![LoadReleaseOption](https://github.com/larspalo/GOODF/assets/6173941/8fa558e1-4f5a-4f3c-a1de-ab182156d0d3)
> 
> 
> 
> You can also fix this without re-loading all the pipes by selecting first attack in the first troublesome pipe and edit its attack properties to not load release, then copy that to all other attacks coming from the same directory.
> 
> 
> 
> ![CopyAttackProperties](https://github.com/larspalo/GOODF/assets/6173941/e2c75fb9-d367-4270-b416-8a41b1102a22)
> 
> 
> 
> All this boils down to that one needs to know how the samples are intended to be used to load them properly... LoopAuditioneer might be a nice friend to get to know so that the samples could be examined.


It does not work when I uncheck the box even if I check the "load any one attack" box. 

In any stop I've created when I play I have notes randomly staying for 4-5 seconds ..

![image](https://github.com/larspalo/GOODF/assets/20370038/67e94e3f-90f6-4ae5-8c9e-1e750192566f)

**Reply by larspalo** [reply]
@Jean-Gabriel1982 The first picture is for when you create a _new_ stop/rank with attack samples that doesn't contain any release in them (that is: in the attack samples - the pipe is almost certainly supposed to have separate releases instead). The second image I posted is for if you need to change the "Load release" property in one or many pipes that have it set wrongly.

From your own image it seems that you indeed have created the pipes with the assumption that there will be a release in the attack samples. This is actually the default value for GOODF since I usually leave the long releases in the attack samples myself when I prepare the samples...

The technical explanation of what you're experience: you have created pipes that try to load a release from the attack sample even though there is no cue marker and actual release tail in them. GrandOrgue will (when that pipe "release" is randomly selected) actually play back that whole sample from the start as the release. The reason why it won't happen every time is because you have real releases present too! Thus it will randomly play back a false release from the attack sample that it shouldn't even try to do.

So, the fix is to tell GrandOrgue not to try to load any release from the attack sample! You do that by setting the Pipe999LoadRelease=N and possibly Pipe999Attack999LoadRelease=N for each attack present in the pipe. Again, this is accomplished **either** by creating the pipes without having the "Load release in attack" checkbox checked, **or**, if you don't want to re-create all the pipes (which is done by first pushing the "Reset/Clear all pipes" and then the "Create new pipes by MIDI nbr from..." _with_ the "Load release in attack" checkbox _unchecked_) follow my suggestion for the second image and the instructions given earlier.

**Reply by Jean-Gabriel1982** [reply]
@larspalo ok i did that and it seems everything is good now but I have many Alerts in GO when I open the organ

![image](https://github.com/larspalo/GOODF/assets/20370038/2f873de7-d9e1-49e5-94d1-266c588108bf)

**Reply by larspalo** [reply]
The warnings are related to a settings file (.cmb), not to the .organ file per se. Might I dare to guess that you've used the same church name (and maybe even .organ file name) as one that you already had registered in GO since before...

Depending on exactly what changes you've made, the settings file might not be compatible with it anymore, but it's generally nothing to worry about (if you know what you're doing, that is). Just ignore the warnings and adjust the settings to your need before saving them again.

But, generally, if you create a new odf, it might be useful to give it another church name (and possibly the odf file name too) so that it won't be confused with older saved presets you might have on your system.

**Comment by Bumblebee001** [comment]
Download and install Loopsuditioneer by Lars himself. Look into the folder
of the main files and see whether there are markers and loops.
